--- configure.in.orig	2012-03-28 18:46:08.163793569 +0530
+++ configure.in	2012-03-28 18:24:22.868791681 +0530
@@ -8086,6 +8086,31 @@ MOZ_TREE_PIXMAN=,
 MOZ_TREE_PIXMAN=1 )
 
 
+dnl ===== Prabu added SGX ======
+SGX_VERSION=1.6.16.4117
+PKG_CHECK_MODULES(SGXCONFIGURO, sgx-null-gles2 >= $SGX_VERSION)
+AC_SUBST(SGXCONFIGURO_CFLAGS)
+AC_SUBST(SGXCONFIGURO_LIBS)
+
+MOZ_QT_LIBS="$SGXCONFIGURO_LIBS $MOZ_QT_LIBS"
+MOZ_QT_CFLAGS="$SGXCONFIGURO_CFLAGS $MOZ_QT_CFLAGS"
+CFLAGS="$SGXCONFIGURO_CFLAGS $CFLAGS"
+LDFLAGS="$SGXCONFIGURO_LIBS $LDFLAGS"
+MOZ_CAIRO_CFLAGS="$SGXCONFIGURO_CFLAGS $MOZ_CAIRO_CFLAGS"
+AC_MSG_CHECKING([Checking for SGX GLES2 compile and link])
+AC_TRY_LINK([#include <GLES2/gl2.h>],
+    { \
+        /* Taken from sgxperf */ \
+        GLuint uiFragShader = glCreateShader(GL_FRAGMENT_SHADER); \
+        return 0; \
+    } \
+    , [ ogles2=yes ], [ ogles2=no ])
+if test "$ogles2" != "yes"; then
+    AC_MSG_ERROR([GLES2 program could not be built. Error!])
+fi
+
+
+
 # Check for headers defining standard int types.
 MOZ_CHECK_HEADERS(stdint.h inttypes.h sys/int_types.h)
 
